### PR TITLE
Oauth2 updates, Guestbook

### PR DESCRIPTION
**Pull request by lesv**
OAuth2
- suggestions from Tim (nits, Guava)
- Suggestions from Brett (HelloServlet now uses User)

Guestbook
- Initial Checkin, updated README

Remove .gitignore where we found it.

### CODE REVIEW

**Comment by lesv** [comment]
@tswast PTAL

**Comment by tmatsuo** [comment]
Possible to add tests?

File: appengine/guestbook-objectify/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tswast) [inline appengine/guestbook-objectify/pom.xml]: License Header + xmlns attributes are optional, so can be removed.

File: appengine/guestbook-objectify/src/main/java/com/example/guestbook/Greeting.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.googlecode.objectify.annotation.Parent;
+
+import java.lang.String;
+import java.util.Date;
+
+/**
+ * The @Entity tells Objectify about our entity.  We also register it in {@link OfyHelper}
+ * Our primary key @Id is set automatically by the Google Datastore for us.
+ *
+ * We add a @Parent to tell the object about its ancestor. We are doing this to support many
```
Comment (tswast) [inline appengine/guestbook-objectify/src/main/java/com/example/guestbook/Greeting.java]: Missing `<p>` tag.

File: appengine/guestbook-objectify/src/main/java/com/example/guestbook/Greeting.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * guestbooks.  Objectify, unlike the AppEngine library requires that you specify the fields you
+ * want to index using @Index.  Only indexing the fields you need can lead to substantial gains in
+ * performance -- though if not indexing your data from the start will require indexing it later.
+ *
+ * NOTE - all the properties are PUBLIC so that can keep the code simple.
```
Comment (tswast) [inline appengine/guestbook-objectify/src/main/java/com/example/guestbook/Greeting.java]: Missing `<p>` tag.

File: appengine/guestbook-objectify/src/main/webapp/guestbook.jsp [hunk 4]
```diff
@@ -0,0 +1,106 @@
+<%-- //[START all]--%>
```
Comment (tswast) [inline appengine/guestbook-objectify/src/main/webapp/guestbook.jsp]: License Header.

**Comment by tswast** [comment]
Minor nits. LGTM.

Agreed with @tmatsuo tests would be good, though I know we are in a rush.

**Pull request merged**